sdk/reqwest/src/gateway/gateway_tx.rs

```diff
@@ -1,119 +1,50 @@
-use anyhow::{anyhow, Result};
-use itertools::Itertools;
-use multiversx_sdk::data::{
-    address::Address,
-    network_config::NetworkConfig,
-    transaction::{
-        ArgCreateTransaction, ResponseTxCost, SendTransactionResponse, SendTransactionsResponse,
-        Transaction, TransactionInfo, TransactionOnNetwork, TransactionProcessStatus,
-        TransactionStatus, TxCostResponseData,
+use anyhow::Result;
+use multiversx_sdk::{
+    data::{
+        address::Address,
+        network_config::NetworkConfig,
+        transaction::{
+            ArgCreateTransaction, Transaction, TransactionOnNetwork, TxCostResponseData,
+        },
+        vm::{VmValueRequest, VmValuesResponseData},
+    },
+    gateway::{
+        GetTxCost, GetTxInfo, GetTxProcessStatus, GetTxStatus, SendMultiTxRequest, SendTxRequest,
+        VMQueryRequest,
     },
-    vm::{ResponseVmValue, VmValueRequest, VmValuesResponseData},
 };
 
 use super::GatewayProxy;
 
-const COST_TRANSACTION_ENDPOINT: &str = "transaction/cost";
-const SEND_TRANSACTION_ENDPOINT: &str = "transaction/send";
-const SEND_MULTIPLE_TRANSACTIONS_ENDPOINT: &str = "transaction/send-multiple";
-const GET_TRANSACTION_INFO_ENDPOINT: &str = "transaction/";
-const WITH_RESULTS_QUERY_PARAM: &str = "?withResults=true";
-const VM_VALUES_ENDPOINT: &str = "vm-values/query";
-
 impl GatewayProxy {
     // request_transaction_cost retrieves how many gas a transaction will consume
     pub async fn request_transaction_cost(&self, tx: &Transaction) -> Result<TxCostResponseData> {
-        let endpoint = self.get_endpoint(COST_TRANSACTION_ENDPOINT);
-        let resp = self
-            .client
-            .post(endpoint)
-            .json(tx)
-            .send()
-            .await?
-            .json::<ResponseTxCost>()
-            .await?;
-
-        match resp.data {
-            None => Err(anyhow!("{}", resp.error)),
-            Some(b) => Ok(b),
-        }
-    }
-
-    async fn get_transaction_info_internal(
-        &self,
-        hash: &str,
-        with_results: bool,
-    ) -> Result<TransactionOnNetwork> {
-        let mut endpoint = GET_TRANSACTION_INFO_ENDPOINT.to_string() + hash;
-
-        if with_results {
-            endpoint += WITH_RESULTS_QUERY_PARAM
-        }
-
-        let endpoint = self.get_endpoint(endpoint.as_str());
-        let resp = self
-            .client
-            .get(endpoint)
-            .send()
-            .await?
-            .json::<TransactionInfo>()
-            .await?;
-
-        match resp.data {
-            None => Err(anyhow!("{}", resp.error)),
-            Some(b) => Ok(b.transaction),
-        }
+        self.request(GetTxCost(tx)).await
     }
 
     // get_transaction_info retrieves a transaction's details from the network
     pub async fn get_transaction_info(&self, hash: &str) -> Result<TransactionOnNetwork> {
-        self.get_transaction_info_internal(hash, false).await
+        // self.get_transaction_info_internal(hash, false).await
```

Comment:
were comments from line 27 and line 36 intentionally left?

Reply:
Nope